### ISSUE TITLE
username variable, for my new/old API..

### ISSUE THREAD

**Issue by shaggyze** [ValerioLyndon/Theme-Customiser#71]
### Has this question been asked already?

- [X] I have searched the existing issues and this is a new question.

### What did you want to know/discuss?

recently updated to v0.4, saw you added

Change theme options
Your MyAnimeList Username

is username replacement global? so I could use it in a api/url like https://shaggyze.website/msa/usercover?u=ShaggyZE for covers or other multiple API generated css?

[css replacements](https://shaggyze.website/msa/usercss?u=ShaggyZE&t=manga&q=%7Btype%7D%2C%20%7Brank%7D%2C%20%7Bid%7D%2C%20%7Bstatus%7D%2C%20%7Bscore%7D%2C%20%7Btags%7D%2C%20%7Bis_rereading%7D%2C%20%7Bnum_read_chapters%7D%2C%20%7Bnum_read_volumes%7D%2C%20%7Bcreated_at%7D%2C%20%7Bmanga_title%7D%2C%20%7Bmanga_english%7D%2C%20%7Bmanga_num_chapters%7D%2C%20%7Bmanga_num_volumes%7D%2C%20%7Bmanga_publishing_status%7D%2C%20%7Bmanga_id%7D%2C%20%7Bmanga_magazines%7D%2C%20%7Bmanga_total_members%7D%2C%20%7Bmanga_total_scores%7D%2C%20%7Bmanga_score_val%7D%2C%20%7Bmanga_score_diff%7D%2C%20%7Bmanga_popularity%7D%2C%20%7Bgenres%7D%2C%20%7Bdemographics%7D%2C%20%7Btitle_localized%7D%2C%20%7Bmanga_url%7D%2C%20%7Bmanga_image_path%7D%2C%20%7Bis_added_to_list%7D%2C%20%7Bmanga_media_type_string%7D%2C%20%7Bstart_date_string%7D%2C%20%7Bfinish_date_string%7D%2C%20%7Bmanga_start_date_string%7D%2C%20%7Bmanga_end_date_string%7D%2C%20%7Bdays_string%7D%2C%20%7Bretail_string%7D%2C%20%7Bpriority_string%7D%2C%20%7Bnotes%7D%2C%20%7Beditable_notes%7D%2C%20%7Bsynopsis%7D)

either way, can you help me with the code to add it and others to my [Zowie](https://shaggyze.github.io/Theme-Customiser/theme?t=https://shaggyze.github.io/Themes/Zowie%20Theme/zowie.json&c=https://shaggyze.github.io/Themes/customiser.json) customiser, is working on my modified [Clarity](https://myanimelist.net/mangalist/ShaggyZE) if its hardcoded

getting a json.parse error atm "Error text: SyntaxError: Bad control character in string literal in JSON at position 2295 (line 73 column 35)"

trying to replace the need of things like my or your scraper for files like https://dl.dropboxusercontent.com/s/4v200vpul24zb4t/maltags6.css so it's auto generated without needing to host/edit/update it or resorting to using all anime/manga...

**Comment by ValerioLyndon** [comment]
Hi Shaggy, did you figure this out eventually? Closing for now as I see your linked Zowie page has a working username option so I assume it is, but please correct me if I am wrong.

As for the username replacement, yes it's usable for anything. It's just a find/replace on the CSS so it works with anything.

**Comment by shaggyze** [comment]
@ValerioLyndon No, I gave up on it after a bit, for some reason the USERNAME doesn't get replaced, pretty sure its just something I'm doing wrong, also need to update to the latest version sometime too

![Image](https://github.com/user-attachments/assets/52724d85-26cb-4133-bb2f-00d7665e0acd)

**Comment by ValerioLyndon** [comment]
I believe this is caused by the `default` value not matching the value being replaced. Whenever the value in the UI matches the default value, replacements are skipped. I forget why it was coded this way, but I think it had to do with not cluttering up saved settings with the entire theme options when all the values were defaulted. I meant to change this ages ago because I also find this behaviour highly counter-intuitive.

The point is, right now `default` values simply auto-fill the UI with whatever template value you want replaced. For some fields, the correct choice may be to not have a `default` so that the field remains blank in the UI. This way, the replacements will kick in once anything has been typed. Since you have a `default` of `ShaggyZE`, the replacements are not being applied any time you put in your username.

Example:
```
"username": {
	"name": "Your MyAnimeList Username",
	"description": "Must be entered exactly as it appears on MyAnimeList.",
	"type": "text",
	"replacements": [
		["USERNAME", "{{{insert}}}"]
	],
	"default": "USERNAME"
}
```
```
"username": {
	"name": "Your MyAnimeList Username",
	"description": "Must be entered exactly as it appears on MyAnimeList.",
	"type": "text",
	"replacements": [
		["USERNAME", "{{{insert}}}"]
	]
}
```

As a side note it also looks like the URL is malformed:
```
… https://shaggyze.website/msa/usercss?u=USERNAMEq=body…
```
Should probably have an `&` after USERNAME:
```
… https://shaggyze.website/msa/usercss?u=USERNAME&q=body…
```

**Comment by shaggyze** [comment]
oops, okay, thanks for looking into it more, will try it soon as I get a chance.
